### DISCUSSION TITLE
Is there any options for using it in a ShareLink?

### DISCUSSION THREAD

**Discussion by vicvoh** [dagronf/QRCode#56] [Q&A] [answered]
I'm now trying to make a ShareLink for the QR-code, but anything I try just doesn't seem to work :( So all I need is code for a ShareLink with this exact QR-code generator, thanks!

**Comment by dagronf** [comment]
@vicvoh What is it you’re trying to share? The image of a QR code, or the QR code definition?

can you please describe exactly what it is you’re trying to do/share? What’s the use case?

**Reply by vicvoh** [reply]
> @vicvoh What is it you’re trying to share? The image of a QR code, or the QR code definition?
> 
> 
> 
> can you please describe exactly what it is you’re trying to do/share? What’s the use case?
> 
> 

I'm trying to share the QR-code **image** via. ShareLink, if you know any options to share an image of a QR-code PNG, PDF and SVG, please let me know😊

- I really love the QR-code generator you've made, thank you so much ☺️

**Comment by dagronf** [comment]
@vicvoh You can use the library to generate an image of the QR code and then use the ShareLink to share the image?

```swift
import SwiftUI
import QRCode

struct ContentView: View {
   let doc = QRCode.Document()

   var body: some View {
      NavigationStack {
         QRCodeDocumentUIView(document: doc)
            .padding()
            .toolbar {
               let qrImage = try! doc.imageUI(CGSize(width: 600, height: 600), label: Text("My QR Code"))
               ShareLink(
                  item: qrImage,
                  preview: SharePreview("My QR Code", image: qrImage)
               )
            }
      }
   }
}

#Preview {
   ContentView()
}
```

**Reply by dagronf** [reply]
@vicvoh If you update to version 21.2.0 there's now a [basic ShareLink demo app](https://github.com/dagronf/QRCode/tree/main/Demo/ShareLink%20Demo) in the `Demos` folder.

You can see the relevant code [here](https://github.com/dagronf/QRCode/blob/main/Demo/ShareLink%20Demo/ShareLink%20Demo/ContentView.swift)

**Comment by vicvoh** [comment]
Thank you for the code, it's working nice, but let's say the user has just styled their QR-code, and they want to share it. To style the QR-code, I'm using your `QRCodeViewUI()` to show the QR-code and it's style, but I can't find any way to convert a `View` into an image to share it in a `ShareLink()` _Sorry for the wrong message I gave you_

**Comment by vicvoh** [comment]
```swift
func generateQRCode(data: String) -> Data? {
        let filter = CIFilter.qrCodeGenerator()
        guard let data = data.data(using: .ascii, allowLossyConversion: false) else { return nil }
        filter.message = data
        guard let ciimage = filter.outputImage else { return nil }
        let transform = CGAffineTransform(scaleX: 16, y: 16)
        let scaledCIImage = ciimage.transformed(by: transform)
        let uiimage = UIImage(ciImage: scaledCIImage)
        return uiimage.pngData()!
    }
``` 
This is the old QR-code generator I used, but it was really bad, since it didn't have any styling and no support for other languages.
Then I came across yours wich is the dream QR-code generator, but I can't share the styled QR-code.
Maybe you could make a `func()` That requires the developer to insert a String, and style commands. Then it'll output image data to be used.

**Comment by dagronf** [comment, marked as answer]
It already has many functions for this on the QRCode.Document object for exporting images, jpeg, png, svg, pdf and vector types.

https://github.com/dagronf/QRCode?tab=readme-ov-file#generating-a-styled-image

Look at the examples given in the documentation for generating image data from a qr code

https://github.com/dagronf/QRCode?tab=readme-ov-file#styling-your-qr-code

The documentation also gives examples of the  more modern 'builder' style interface for easily generating an image

https://github.com/dagronf/QRCode?tab=readme-ov-file#qrcode-builder

Also there are plenty of real-world examples of building qr code images in the `Examples` subfolder

https://github.com/dagronf/QRCode/tree/main/Examples
